libass/ass_render.c

```diff
@@ -1495,25 +1497,45 @@ static void measure_text(ASS_Renderer *render_priv)
     int cur_line = 0;
     double scale = 0.5 / 64;
     int max_asc = 0, max_desc = 0;
+    double max_border_y = 0, max_border_x = 0;
     for (int i = 0; i < text_info->length; i++) {
         if (text_info->glyphs[i].linebreak) {
             max_asc = max_desc = 0;
+            max_border_y = 0;
             scale = 0.5 / 64;
             cur_line++;
         }
         GlyphInfo *cur = text_info->glyphs + i;
         max_asc  = FFMAX(max_asc,  cur->asc);
         max_desc = FFMAX(max_desc, cur->desc);
+        max_border_y = FFMAX(max_border_y, cur->border_y);
+        max_border_x = FFMAX(max_border_x, cur->border_x);
         if (cur->symbol != '\n' && cur->symbol != 0)
             scale = 1.0 / 64;
         if (i == text_info->length - 1 || text_info->glyphs[i + 1].linebreak) {
             text_info->lines[cur_line].asc  = scale * max_asc;
             text_info->lines[cur_line].desc = scale * max_desc;
             text_info->height += scale * max_asc + scale * max_desc;
+            // For *VSFilter compatibility do biased rounding on max_border*
+            // https://github.com/Cyberbeing/xy-VSFilter/blob/xy_sub_filter_rc4@%7B2020-05-17%7D/src/subtitles/RTS.cpp#L1465
+            text_info->border_bot =
+                (int) (render_priv->border_scale * max_border_y + 0.5);
+            if (cur_line == 0)
+                text_info->border_top = text_info->border_bot;
         }
     }
     assert(cur_line == text_info->n_lines - 1);
     text_info->height += cur_line * render_priv->settings.line_spacing;
+    // VSFilter does not care if the max x-border is actuallly close to an edge
```

Comment:
`actuallly`

Reply:
I've reworded the whole comment and shortened it to one line